src/osd/OSD.cc

```diff
@@ -7458,6 +7467,13 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
 	do_restart = true;
       }
     }
+    if (osdmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS) !=
+	newmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
+      dout(10) << __func__ << " REQUIRE_LUMINOUS flag changed in " << newmap->get_epoch()
+	       << dendl;
+      clear_pg_stat_queue();
```

Comment:
quote from https://github.com/liewegas/ceph/pull/35#discussion_r116745177

> this isn't quite sufficient since the pgs osdmaps lag and they will still queue themselves...